homeassistant/components/dexcom/sensor.py

```diff
@@ -36,11 +39,15 @@ class DexcomSensorEntity(CoordinatorEntity, SensorEntity):
     """Base Dexcom sensor entity."""
 
     def __init__(
-        self, coordinator: DataUpdateCoordinator, username: str, key: str
+        self, coordinator: DataUpdateCoordinator, username: str, entry_id: str, key: str
     ) -> None:
         """Initialize the sensor."""
         super().__init__(coordinator)
         self._attr_unique_id = f"{username}-{key}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, entry_id)},
```

Comment:
This is close to the edge of my understanding on unique IDs but I think this approach might not be reliable enough in certain situations.

- if you remove and re-add the entities/device, it might not come back with the same unique id, resulting in automations/scripts all having to be fixed.

If dexcom only ever has one active 'device' per username, then the username might be enough.  In that case the HA 'device' is really the user account rather than the physical glucose monitor. This could even be preferred in the case of having to replace a broken monitor (new monitor but same human).